platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/ConfigurationCacheRelativePaths.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cc.impl
+
+import org.gradle.initialization.layout.BuildTreeLocations
+import org.gradle.internal.service.scopes.Scope
+import org.gradle.internal.service.scopes.ServiceScope
+import org.gradle.util.internal.GFileUtils.relativePathOf
+import java.io.File
+
+
+@ServiceScope(Scope.BuildTree::class)
+internal
+interface ConfigurationCacheRelativePaths {
```

Comment:
I think class level javadoc here would be useful.

Why does configuration cache need relative paths? When should I relativize paths with this class? What is the base dir? Is it important?

Reply:
Additionally, if base dir is an important and relevant concept for the interface (not just the default impl), there should probably be a getter for it on the interface.

Reply:
>  if base dir is an important and relevant concept for the interface (not just the default impl), there should probably be a getter for it on the interface.


In the other hand, if the goal of this service is to allow clients to avoid absolute locations, exposing the base location seems to be at odds with that goal. Unless there is a use case for accessing the base location, I would avoid exposing it.

Reply:
Thank you for the feedback. It made me change my mind about introducing this class. I've removed it from the changeset.

My initial idea was to start consolidating places where CC already tries to support relocatability in the form of relativizing files. At least it seemed suspicious that we have scattered places that relativize paths using `GFileUtils.relativePathOf`always with the build-tree root base. Looking at it again, it's questionable whether we should be doing it exactly that way everywhere, so I moved the changes out of the PR.